bento_lib/auth/middleware.py

```diff
@@ -0,0 +1,90 @@
+import json
+import requests
+import jwt
+import time
+
+from threading import Thread
+from flask import request, g
+from jwt.algorithms import RSAAlgorithm
+
+
+class AuthxFlaskMiddleware():
+    def __init__(self, oidc_iss="https://localhost/auth/realms/realm", client_id="abc123", oidc_alg="RS256"):
+        print('authx middleware initialized')
+
+        self.oidc_issuer = oidc_iss
+        self.client_id = client_id
+        self.oidc_alg = oidc_alg
+
+        self.oidc_wellknown_path = self.oidc_issuer + "/protocol/openid-connect/certs"
```

Comment:
this should probably be pulled from the well-known OIDC config path for the IdP in question via a request rather than a direct hard-coding of the certs path, since that would change from IdP to IdP